edgexfoundry/edgex-compose
# feat: Ensure all services, that should, restart after reboot.

closes #184

Have to use `restart: always` to ensure container restart after reboot. 







**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-compose/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

  **Docs not impacted**


## Testing Instructions
1. run the default secure compose file, i.e. `make run` from repo root folder
2. run `make get-token` and use token to to do GET to https://localhost:8443/core-metadata/api/v2/ping
   - Verify 200 response code.
   - Save token for use later
3. run `make get-consul-acl-token` and use token to login to Consul at localhost:8500 in browser
4. Navigate to `http://localhost:8500/ui/dc1/kv/edgex/appservices/2.0/app-rules-engine/Writable/LogLevel/edit`
5. Change value to `DEBUG`
6. run `make portainer`
7. Open localhost:9000 in browser and set password if prompted.
8. Navigate to viewing containers and then display logs for ` edgex-app-rules-engine`
   - Verify data is flowing, i.e. logs show events being processed
9. Reboot the system
10. Open localhost:9000 in browser and verify all containers have started.
11. Open Consul at localhost:8500 in browser and navigate to `http://localhost:8500/ui/dc1/kv/edgex/appservices/2.0/app-rules-engine/Writable/LogLevel/edit`
   - Note you are still logged in. i.e. didn't have to use access token
12. Use API Token from above to to do GET to https://localhost:8443/core-metadata/api/v2/ping
   - Verify 200 response code.
